### PR TITLE
fix(identify): re-render if idx.unknown.user message

### PR DESCRIPTION
**Pull request by sherizada-okta**
## Description:
- the bug: Identify form was stuck in saving mode if the user entered the same identifier twice in a row because form does not re-render if previous and current API responses are identical
- the fix: re-render if API response contains a message with "idx.unknown.user" key

## PR Checklist

- [x] Have you verified the basic functionality for this change?
- [x] Did you add tests, as appropriate, following our [Automated Test guidelines](https://oktawiki.atlassian.net/wiki/spaces/eng/pages/2676497890/Automated+Testing+in+the+Signin+Widget)?
- [x] Did you follow our [Security Best Practices](https://oktawiki.atlassian.net/wiki/display/eng/Security+Best+practices)?
- [ ] Did you verify the change by running [downstream monolith artifact](https://oktawiki.atlassian.net/wiki/spaces/eng/pages/102897979/Sign-in+Widget+Development#Sign-inWidgetDevelopment-Instructionstocreateandrunthedownstreamartifact(d16t))? (Provide link to build below)
- [x] Does this PR include noticeable changes to the UI? (If yes, attach screenshots/video below)

### Issue:

- [OKTA-520158](https://oktainc.atlassian.net/browse/OKTA-520158)

### Screenshot/Video:
Bug:

https://user-images.githubusercontent.com/71431120/184957055-4cadbe5f-9bff-4934-a148-0cc91d316d13.mov

Fix:

https://user-images.githubusercontent.com/71431120/184957287-e3c4ba56-187e-4c9e-bac5-fa57dfbb57cb.mov

### CODE REVIEW

File: test/unit/spec/v2/models/AppState_spec.js [hunk 1]
```diff
@@ -187,6 +187,53 @@ describe('v2/models/AppState', function() {
       expect(this.appState.shouldReRenderView(transformedResponse)).toBe(false);
     });
 
+    it('rerender view should be true if idx message contains idx.unknown.user', () => {
```
Comment (lesterchoi-okta) [inline test/unit/spec/v2/models/AppState_spec.js:190]:
```suggestion
    it('should re-render the view if the idx message contains idx.unknown.user', () => {
```

File: src/v2/models/AppState.ts [hunk 2]
```diff
@@ -314,26 +315,31 @@ export default class AppState extends Model {
...
+         */
+        reRender = true;
+      } else if (FORMS_WITH_STATIC_BACK_LINK.includes(this.get('currentFormName'))) {
+        /**
+         * returns true: We want to force reRender if you go back to selection screen from challenge or enroll screen
+         * and re-select the same authenticator for challenge. In this case also new response will be identical
+         * to the old response.
+         */
+        reRender = true;
+      } else if (this.containsMessageWithI18nKey(UNKNOWN_USER_I8N_KEY)) {
```
Comment (lesterchoi-okta) [inline src/v2/models/AppState.ts:333]: Is this the only new case that needs to be re-rendered? i.e., is the `idx.unknown.user` response actually special or is it just the one that was noticed/reported?

**Review by aarongranick-okta — APPROVED** [review]

**Review by lesterchoi-okta — COMMENTED** [review]
I would check to see if matching on `idx.unknown.user` i18n key is the right condition. Have you verified that this doesn't happen on other screens with similar steps-to-reproduce?

File: src/v2/models/AppState.ts (hunk 2 above)
Comment (sherizada-okta) [reply src/v2/models/AppState.ts:333]: It's special: https://okta.slack.com/archives/CCA77QVL2/p1660695611393009.
ftr, more info on this PR for future devs: https://okta.slack.com/archives/CAZH2MWEL/p1660675971972929

**Pull request merged**